controllers/propagator/policy_controller.go

```diff
@@ -122,20 +123,63 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return reconcile.Result{}, err
 			}
 
+			if len(replicatedPlcList.Items) == 0 {
+				log.V(2).Info("No replicated policies to delete.")
+
+				return reconcile.Result{}, nil
+			}
+
+			policiesChan := make(chan policiesv1.Policy, len(replicatedPlcList.Items))
+			deletionResultsChan := make(chan deletionResult, len(replicatedPlcList.Items))
+
+			var numWorkers int
+			if len(replicatedPlcList.Items) > concurrencyPerPolicy {
+				numWorkers = concurrencyPerPolicy
+			} else {
+				numWorkers = len(replicatedPlcList.Items)
+			}
+
+			for i := 0; i < numWorkers; i++ {
+				go plcDeletionWrapper(r, policiesChan, deletionResultsChan)
+			}
+
+			log.V(2).Info("Scheduling work to handle deleting replicated policies for the policy", "policy",
+				instance.GetName())
+
 			for _, plc := range replicatedPlcList.Items {
-				log := log.WithValues("name", plc.GetName(), "namespace", plc.GetNamespace())
+				policiesChan <- plc
+			}
+
+			// Wait for all the deletions to be processed.
+			log.V(2).Info("Waiting for the result of deleting the replicated policies", "count", len(policiesChan))
 
-				log.V(1).Info("Deleting the replicated policy")
+			processedResults := 0
+			failures := 0
+
+			for result := range deletionResultsChan {
+				if result.Err != nil {
+					failures++
+				}
 
-				// #nosec G601 -- no memory addresses are stored in collections
-				err := r.Delete(ctx, &plc)
-				if err != nil && !errors.IsNotFound(err) {
-					log.Error(err, "Failed to delete the replicated policy")
+				processedResults++
 
-					return reconcile.Result{}, err
+				// Once all the decisions have been processed, it's safe to close
+				// the channels and stop blocking in this goroutine.
+				if processedResults == len(replicatedPlcList.Items) {
+					close(policiesChan)
+					close(deletionResultsChan)
+					log.V(2).Info("All replicated policy deletions have been handled\n", "count",
+						len(replicatedPlcList.Items))
 				}
 			}
 
+			if failures > 0 {
+				err := genericError.New("replicated policy deletion failed")
+				log.Error(err, "Not all replicated policies were deleted")
```

Comment:
I think you could make this an info message since the code that encounters the failure logs it.